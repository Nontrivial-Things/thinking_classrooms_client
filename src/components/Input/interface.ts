export default interface SerachInputProps {
  // event: React.MouseEventHandler;
  suggestions: Array<string>;
}
