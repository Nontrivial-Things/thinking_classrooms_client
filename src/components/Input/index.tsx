import React, { FC, useState } from "react";

// import InputProps from "./interface";
import Input from "./styledComponents";
import Row from "../Row/styledComponents";
import SearchIcon from "../../assets/img/icons/search.svg";
import CloseIcon from "../../assets/img/icons/remove-icon.svg";

const SearchInput: FC = () => {
  const [searchTerm, setSearchTerm] = useState("");
  const handleChange = (e: React.FormEvent<HTMLInputElement>) =>
    setSearchTerm(e.currentTarget.value);

  // const clearInput = () => {
  //   setSearchTerm("");
  // };

  return (
    <Row>
      {/* <form action="/" method="get"> */}
      <label htmlFor="header-search" />
      <Input
        type="text"
        placeholder="Szukaj problemÃ³w matematycznych"
        value={searchTerm}
        onChange={handleChange}
        onSubmit={(e) => {
          e.preventDefault();
        }}
      />
      {/* <SearchIcon /> */}
      {/* </form> */}
    </Row>
  );
};

export default SearchInput;
