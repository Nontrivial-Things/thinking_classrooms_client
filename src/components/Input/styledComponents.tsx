import styled from "styled-components";

import { minDevice } from "../../assets/styles/breakpoints";
import {
  white,
  placeholder,
  activeButton,
  lightGrey,
} from "../../assets/styles/colors";
import { ReactComponent as LensIcon } from "../../assets/img/icons/search.svg";
import { ReactComponent as ClearIcon } from "../../assets/img/icons/remove-icon.svg";

const FormWrapper = styled.div`
  width: 100%;
  box-shadow: 0px 4px 15px 0px rgba(22, 28, 86, 0.2);
  @media ${minDevice.desktopTablet} {
    width: 22.5em;
  }
  @media ${minDevice.desktopL} {
    width: 31.4375em;
  }
`;

const Form = styled.form`
  width: 100%;
`;

const Input = styled.input`
  width: 100%;
  height: 3em;
  padding: 0;
  box-sizing: border-box;
  padding: 0 2.5em;
  border-radius: 5px;
  background: ${white};
  border: none;
  font-family: inherit;
  font-size: 0.875em;
  line-height: 1.5em;
  ::placeholder {
    color: ${placeholder};
  }
  &:focus-visible {
    outline: none;
  }
`;

const Label = styled.label`
  position: relative;
`;

const SearchIcon = styled(LensIcon)`
  stroke: ${activeButton};
  left: 0.5em;
  bottom: 0;
  position: absolute;
`;

const RemoveIcon = styled(ClearIcon)`
  fill: ${lightGrey};
  width: 1em;
`;

const SuggestionList = styled.ul`
  position: relative;
  width: 100%;
  max-height: 22.5em;
  border-top: 1px solid;
  border-top-color: ${activeButton};
  overflow: scroll;
  @media ${minDevice.desktopL} {
    max-height: 30.5625em;
  }
`;

const Suggestion = styled.li`
  font-size: 0.875em;
  box-sizing: border-box;
  padding: 0.5em;
  width: 100%;
  display: flex;
  align-items: center;
  &:hover {
    /* filter: drop-shadow(0px 4px 15px red); */
    background: rgba(22, 28, 86, 0.1);
  }
  &:focus-visible,
  &:focus,
  &:active {
    background: rgba(22, 28, 86, 0.1);
  }
`;

const Button = styled.button`
  width: auto;
  border: none;
  right: 0.5em;
  top: calc(50% - 1.5em / 2 - 0.5px);
  bottom: 0;
  position: absolute;
  cursor: pointer;
  background: transparent;
`;

export {
  Form,
  Input,
  Label,
  SuggestionList,
  Suggestion,
  SearchIcon,
  RemoveIcon,
  FormWrapper,
  Button,
};
