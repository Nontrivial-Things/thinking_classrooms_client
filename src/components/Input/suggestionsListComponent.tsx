import { FC } from "react";
import { SuggestionsProps } from "./interface";
import { SuggestionList, Suggestion } from "./styledComponents";
import { ReactComponent as Clipboard } from "../../assets/img/icons/clipboard.svg";

const SuggestionsListComponent: FC<SuggestionsProps> = ({
  filteredSuggestions,
  activeSuggestionIndex,
  chooseSuggestion,
}) => {
  return filteredSuggestions.length ? (
    <SuggestionList>
      {filteredSuggestions.map((suggestion: string, index: number) => {
        // if (index === activeSuggestionIndex) {
        // }
        return (
          <Suggestion key={suggestion} onClick={(e) => chooseSuggestion(e)}>
            <Clipboard style={{ paddingRight: "0.5em" }} />
            <span>{suggestion}</span>
          </Suggestion>
        );
      })}
    </SuggestionList>
  ) : (
    <div>
      <em>Brak wynik√≥w</em>
    </div>
  );
};

export default SuggestionsListComponent;
