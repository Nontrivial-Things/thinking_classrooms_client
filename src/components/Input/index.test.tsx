import { render, screen, fireEvent, within } from "@testing-library/react";

import Input from "./index";

describe("Input component", () => {
  it("should allow letters to be inputted", () => {
    render(
      <Input
        suggestions={[
          "Alligator",
          "Bask",
          "Crocodilian",
          "Death Roll",
          "Eggs",
          "Jaws",
          "Reptile",
          "Solitary",
          "Tail",
          "Wetlands",
        ]}
      />
    );

    let input = screen.getByLabelText("search-input") as HTMLInputElement;
    expect(input.value).toBe("");
    fireEvent.change(input, { target: { value: "Good Day" } });
    expect(input.value).toBe("Good Day");
  });

  it("should show suggestions list matching input target", () => {
    render(
      <Input
        suggestions={[
          "Alligator",
          "Bask",
          "Crocodilian",
          "Death Roll",
          "Eggs",
          "Jaws",
          "Reptile",
          "Solitary",
          "Tail",
          "Wetlands",
        ]}
      />
    );
    let input = screen.getByLabelText("search-input") as HTMLInputElement;

    expect(input.value).toBe("");
    fireEvent.change(input, { target: { value: "e" } });
    expect(input.value).toBe("e");

    const list = screen.getByRole("list");
    const { getAllByRole } = within(list);
    const items = getAllByRole("listitem");
    expect(items.length).toBe(4);
  });

  it("should show in input field chosen suggestion", () => {
    render(
      <Input
        suggestions={[
          "Alligator",
          "Bask",
          "Crocodilian",
          "Death Roll",
          "Eggs",
          "Jaws",
          "Reptile",
          "Solitary",
          "Tail",
          "Wetlands",
        ]}
      />
    );
    let input = screen.getByLabelText("search-input") as HTMLInputElement;

    expect(input.value).toBe("");
    fireEvent.change(input, { target: { value: "All" } });
    expect(input.value).toBe("All");

    const list = screen.getByRole("list");
    const { getAllByRole } = within(list);
    const items = getAllByRole("listitem");

    expect(items.length).toBe(1);

    const item = screen.getByText("Alligator").textContent;

    fireEvent.change(input, { target: { value: item } });
    expect(input.value).toBe("Alligator");
  });

  it("shouldn't show remove button in input field", () => {
    render(
      <Input
        suggestions={[
          "Alligator",
          "Bask",
          "Crocodilian",
          "Death Roll",
          "Eggs",
          "Jaws",
          "Reptile",
          "Solitary",
          "Tail",
          "Wetlands",
        ]}
      />
    );
    let input = screen.getByLabelText("search-input") as HTMLInputElement;

    expect(input.value).toBe("");
    const button = screen.queryByRole("button", { name: /Remove button/i });

    expect(button).not.toBeInTheDocument();
  });

  it("should show remove button and after click remove text from input field", () => {
    render(
      <Input
        suggestions={[
          "Alligator",
          "Bask",
          "Crocodilian",
          "Death Roll",
          "Eggs",
          "Jaws",
          "Reptile",
          "Solitary",
          "Tail",
          "Wetlands",
        ]}
      />
    );
    let input = screen.getByLabelText("search-input") as HTMLInputElement;

    expect(input.value).toBe("");
    fireEvent.change(input, { target: { value: "B" } });
    expect(input.value).toBe("B");

    const button = screen.queryByRole("button", {
      name: /Remove button/i,
    }) as HTMLButtonElement;

    expect(button).toBeInTheDocument();

    fireEvent.click(button);

    expect(input.value).toBe("");
  });
});
