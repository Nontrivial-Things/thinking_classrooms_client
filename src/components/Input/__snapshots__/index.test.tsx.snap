// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should key press correctly 1`] = `
<ul
  class="sc-gKclnd bPwDyF"
/>
`;
